File: Spoj/EDIST.cpp
#include <iostream>
#include <string>

using namespace std;

int getEditDistance(string A, string B)
{
    int len1 = A.length();
    int len2 = B.length();
    int dp[len1 + 1][len2 + 1];
    for (int i = 0; i <= len1; i++)
    {
        for (int j = 0; j <= len2; j++)
        {
            if (i == 0)
            {
                dp[i][j] = j;
            }
            else if (j == 0)
            {
                dp[i][j] = i;
            }
            else if (A[i-1] == B[j-1])
            {
                dp[i][j] = dp[i - 1][j - 1];
            }
            else
            {
                dp[i][j] = min(dp[i - 1][j], min(dp[i][j - 1], dp[i - 1][j - 1])) + 1;
            }
        }
    }
    return dp[len1][len2];
}

void solve()
{
    int N;
    cin >> N;
    while (N--)
    {
        string A, B;
        cin >> A >> B;
        cout << getEditDistance(A, B) << "\n";
    }
}

int main()
{
    freopen("/Users/gopaljaiswal/project/frameworks/Anaconda_python/Comp_Prog/"
            "input.txt",
            "r", stdin);
    freopen("/Users/gopaljaiswal/project/frameworks/Anaconda_python/Comp_Prog/"
            "output.txt",
            "w", stdout);

    solve();
    return 0;
}

File: Spoj/EDIST.py


def editDistDP(s1, s2, m, n):
    dp = [[0 for i in range(n+1)] for j in range(m+1)]
    for i in range(0, m+1):
        for j in range(0, n+1):
            if i == 0:
                dp[i][j] = j
            elif j == 0:
                dp[i][j] = i
            elif s1[i-1] == s2[j-1]:
                dp[i][j] = dp[i-1][j-1]
            else:
                dp[i][j] = 1 + min(dp[i][j-1], dp[i-1][j], dp[i-1][j-1])
    print dp[m][n]


T = input()
for i in range(0, T):
    s1 = raw_input()
    s2 = raw_input()
    editDistDP(s1, s2, len(s1), len(s2))
